source/appModules/teamtalk5classic.py

```diff
@@ -0,0 +1,220 @@
+ID_UNBANNEDLIST = 1327
+ID_TOOLBAR = 59392
+ID_VU = 1004
+
+
+class TTToolBar(IAccessible):
+	def initOverlayClass(self):
+		n = self.childCount
+		# TeamTalk version-specific toolbar item names by toolbar item count.
+		self.names = []
+		for lst in _msgNames.values():
+			if len(lst) == n:
+				self.names = lst
+				break
+
+	def getToolBar(self):
+		# First lastChild is a Window object, then the client toolbar object.
+		try:
+			tb = api.getForegroundObject().lastChild.lastChild
+		except AttributeError:
+			tb = None
+		return tb
+
+	def name(self, itemID):
+		"""The name of one item.
+		"""
+		if itemID <= 0:
+			return ""
+		# ChildIDs are 1-based, but Python arrays are 0-based.
+		try:
+			name = self.names[itemID - 1]
+		except IndexError:
+			return ""
+		if name == "sep":
+			return ""
+		return name
+
+
+class TTToolBarItem(IAccessible):
+	def event_valueChange(self):
+		speech.speakObject(self)
+
+	def _get_name(self):
+		return self.parent.info.name(self.IAccessibleChildID)
+
+
+class AppModule(appModuleHandler.AppModule):
+	def event_NVDAObject_init(self, obj):
+		# For some TeamTalk versions, the incoming chat control does not
+		# return the correct _isWindowUnicode flag.
+		if obj.windowClassName == "RichEdit20A":
+			obj._isWindowUnicode = False
+		# Label channel chat in/out controls.
+		if obj.windowControlID == ID_CHATOUT:
+			# Translators: a label for a read only edit control
+			# in the TeamTalk voice conferencing client where incoming chat
+			# messages are displayed.
+			obj.name = _("Channel messages")
+		elif obj.windowControlID == ID_CHATIN:
+			# Translators: a label for the TeamTalk chat entry field
+			obj.name = _("Channel input")
+		# Name the lists in the Banned Users dialog.
+		elif obj.role == controlTypes.ROLE_LIST:
+			if obj.windowControlID == ID_BANNEDLIST:
+				# Translators: Label for a list of banned users in TeamTalk
+				obj.name = _("Banned")
+			elif obj.windowControlID == ID_UNBANNEDLIST:
+				# Translators: Label for a list of unbanned users in TeamTalk
+				obj.name = _("Unbanned")
+
+	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
+		# There is a VU meter progress bar in the main window
+		# that shouldn't be monitored (it presents many spurious updates).
+		if obj.windowClassName == "msctls_progress32" and (
+			obj.name == "VU"
+			or obj.windowControlID == ID_VU
+		):
+			try:
+				clsList.remove(ProgressBar)
+			except ValueError:
+				from logHandler import log
+				log.exception("Couldn't remove VU meter progress bar")
+		# Name toolbar icons.
+		elif obj.windowControlID == ID_TOOLBAR:
+			# Applies to toolbar and its items.
+			if obj.role == controlTypes.ROLE_TOOLBAR:
+				clsList.insert(0, TTToolBar)
+			elif obj.parent and obj.parent.role == controlTypes.ROLE_TOOLBAR:
+				clsList.insert(0, TTToolBarItem)
+
+	@scriptHandler.script(
+		gesture="kb:control+shift+s",
+		# Translators: a gesture description for the TeamTalk voice conferencing client.
+		description=_(
```

Comment:
Still very long. Please shorten this to a max of 3 sentences, where the first setence just says something like: "Announces relevant information from the Team Talk tool bar" or something like that. Also, to more closely reflect how scripts are documented in core, please change "oon frist press" to something like "pressing once"